android/java/org/chromium/chrome/browser/crypto_wallet/fragments/EditVisibleAssetsBottomSheetDialogFragment.java

```diff
@@ -609,66 +609,81 @@ public void onMaybeShowTrashButton(
-                                    } else {
-                                        braveWalletService.removeUserAsset(
-                                                walletListItemModel.getBlockchainToken(),
-                                                (success) -> {
-                                                    if (success) {
-                                                        walletListItemModel.setIsUserSelected(
-                                                                false);
-                                                    }
-                                                    if (isChecked
-                                                            != walletListItemModel
-                                                                       .getIsUserSelected())
-                                                        assetCheck.setChecked(
-                                                                walletListItemModel
-                                                                        .getIsUserSelected());
-                                                });
-                                    }
-                                } else {
-                                    braveWalletService.setUserAssetVisible(
-                                            walletListItemModel.getBlockchainToken(), isChecked,
-                                            success -> {
-                                                if (success) {
-                                                    walletListItemModel.setIsUserSelected(
-                                                            isChecked);
-                                                }
-                                                if (isChecked
-                                                        != walletListItemModel.getIsUserSelected())
-                                                    assetCheck.setChecked(
-                                                            walletListItemModel
-                                                                    .getIsUserSelected());
-                                            });
-                                }
-                                mIsAssetsListChanged = true;
-                            });
+                thisToken.coin, selectedNetwork -> {
+            TokenUtils.isCustomToken(getBlockchainRegistry(), selectedNetwork,
+                                    selectedNetwork.coin, thisToken,
+                                    isCustom -> {
+                // Only show add asset dialog on click when:
+                //    1. It is an ERC721 token
+                //    2. It is a token listed in Registry
+                //    3. It is not user added (i.e. doesn't have a token id)
+                if (thisToken.isErc721 && !isCustom
+                        && (thisToken.tokenId == null || thisToken.tokenId.trim().isEmpty())) {
```

Comment:
I wonder what happens with SOL NFTs with that logic. NTFs there don't have tokenId and I think we mark the isErc721 as well. I think we should add a condition that it's not a SOL token.

Reply:
I see, don't know about SOL. Added now.

Reply:
I don't have any NFTs on SOL Devnet to test, but I guess technically ERC721 is an ETH standard? How is the backend implemented?

Reply:
@yrliou @wchen342 that's how we set it when we add it, so for custom SOL NFTs we for sure have `isErc721 = true` and `isNft = true`. Not sure how backend returns for NFTs that are in the registry.

Reply:
@yrliou we have that when we add custom assets `token.isNft = isErc721;`

Reply:
and it's been added by you @yrliou https://github.com/brave/brave-core/commit/1221807ce94b864848778c5c76149a83e3f0d905

Reply:
I think we are good actually in that place even without a check on SOL as `isErc721` should be false for SOL NFTs, it's only both `isErc721` and `isNft` true for ERC721 custom tokens

Reply:
@SergeyZhukovsky Yes, isNft should be true whenever isErc721 is true on EVM, that's what I added (no SOL cases back then). I was asking the other way around above, when isNft = true, it doesn't guarantee isErc721 will be true.
We don't have Solana NFTs in the registry currently AFAIK, it doesn't make much sense to me anyway because each Solana NFT has it's unique address, so I wouldn't imagine that would be listed in the registry, at least for now.
We will never set isErc721 = true for Solana tokens in the backend, if there is, it's a bug.

Reply:
So basically checking `isErc721` is good I think. Unless we have SOL NFT in registry (which we don't I suppose?)

Reply:
yeah that's right so to sum up:
1. `isNft == true` and `isErc721 ==  false` for SOL NFTs.
2. `isNft == true` and `isErc721 == true` for ERC721.
I think we are good as it is in that case if we don't have any SOL NFTs in the registry.

Reply:
Yeah I have updated the part, it's good now.